code/game/objects/structures/ghost_role_spawners.dm

```diff
@@ -132,14 +133,31 @@
 	if(ishuman(new_spawn))
 		var/mob/living/carbon/human/H = new_spawn
 		H.set_cloned_appearance()
-		H.real_name = H.dna.species.random_name()
+		if(!name)
+			H.real_name = H.dna.species.random_name()
+		else
+			H.real_name = name
+
+/obj/effect/mob_spawn/human/golem/attack_hand(mob/user)
+	if(isgolem(user) && can_transfer)
+		var/transfer = alert("Transfer your soul to [src]? (Warning, your old body will die!)",,"Yes","No")
+		if(!transfer)
+			return
+		log_game("[user.ckey] golem-swapped into [src]")
+		user.visible_message("<span class='notice'>A faint light leaves [user], moving to [src] and animating it!</span>","<span class='notice'>You leave your old body behind, and transfer into [src]!</span>")
+		create(ckey = user.ckey, flavour = FALSE, name = user.real_name)
+		user.death()
+		return
+	..()
+
 
 /obj/effect/mob_spawn/human/golem/adamantine
 	name = "dust-caked golem shell"
 	desc = "A humanoid shape, empty, lifeless, and full of potential."
 	mob_name = "a free golem"
 	anchored = 1
 	density = 1
+	can_transfer = FALSE
```

Comment:
Adamantium golems don't have a ore type though, though that'd be a good second option @octareenroon91 since you can initially break golems down in a gibber.